CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,162 @@
+others. Do not insult or put down other participants.  We will not accept
+harassment or other exclusionary behaviour, such as:
+
+- Violent threats or language directed against another person.
+- Sexist, racist, or otherwise discriminatory jokes and language.
+- Posting sexually explicit or violent material.
+- Posting (or threatening to post) other people's personally identifying information ("doxing").
+- Sharing private content, such as emails sent privately or non-publicly,
+or unlogged forums such as IRC channel history, without the sender's consent.
+- Personal insults, especially those using racist or sexist terms.
+- Unwelcome sexual attention.
+- Excessive profanity. Please avoid swearwords; people differ greatly in their sensitivity to swearing.
+- Repeated harassment of others. In general, if someone asks you to stop, then stop.
+- Advocating for, or encouraging, any of the above behaviour.
+
+
+Diversity Statement
+-------------------
+
+The PySAL project welcomes and encourages participation by everyone. We are
+committed to being a community that everyone enjoys being part of. Although
+we may not always be able to accommodate each individual's preferences, we try
+our best to treat everyone kindly.
+
+No matter how you identify yourself or how others perceive you: we welcome you.
+Though no list can hope to be comprehensive, we explicitly honour diversity in:
+age, culture, ethnicity, genotype, gender identity or expression, language,
+national origin, neurotype, phenotype, political beliefs, profession, race,
+religion, sexual orientation, socioeconomic status, subculture and technical
+ability, to the extent that these do not conflict with this code of conduct.
+
+
+Though we welcome people fluent in all languages, PySAL development is
+conducted in English.
+
+Standards for behaviour in the PySAL community are detailed in the Code of
+Conduct above. Participants in our community should uphold these standards
+in all their interactions and help others to do so as well (see next section).
+
+
+Reporting Guidelines
+--------------------
+
+We know that it is painfully common for internet communication to start at or
+devolve into obvious and flagrant abuse.  We also recognize that sometimes
+people may have a bad day, or be unaware of some of the guidelines in this Code
+of Conduct. Please keep this in mind when deciding on how to respond to a
+breach of this Code.
+
+For clearly intentional breaches, report those to the Code of Conduct committee
+(see below). For possibly unintentional breaches, you may reply to the person
+and point out this code of conduct (either in public or in private, whatever is
+most appropriate). If you would prefer not to do that, please feel free to
+report to the Code of Conduct Committee directly, or ask the Committee for
+advice, in confidence.
+
+You can report issues to the PySAL Code of Conduct committee, at
+***pysal-conduct@googlegroups.com***. Currently, the committee consists of:
```

Comment:
Create new google group?

Reply:
I'd use the same google group

Reply:
I think someone with a concern should have a committee-only communication option. A Google Group is an interesting model, but one that has pros and cons. My understanding is that the group archives correspondence. This means that committee members a few years from now will be able to see the issues posted today.
- This could be good for new members to potentially identify trends in complaints.
- Could be bad if the original submitter assumed their concern would only be visible to the contemporaneous committee members.

One option would be to give someone multiple avenues to submit a concern: pysal-conduct@googlegroups.com, pysal-devs@googlegroups.com and direct email to committee members.

Reply:
I would be plus one on the Gitter to not spread our communication platforms too far. Additionally, it is possible to sent members private messages on Gitter. If your click on the persons picture opening additional information, there is an option to `Chat privately` in the right lower corner. This might be good to explicitly point out, in addition to adding e-mails?

Reply:
I think the problem with having conduct concerns sent to one committee member (via personal email or gitter message) has the drawback of not involving the other committee members. The purpose of having a committee is to ensure transparency (in a private manner) while addressing conduct concerns.

Potential solutions:
- a private google-group / email address to which a concerned party can send a message and the committee members are notified.
- the account/group would be opened and maintained by `PySAL`, not any individual developer or committee member.
- the account must persist from 'generation to generation' of `pysal` devs.
- etc?

I am not so mush against the idea of private email or gitter chat, but it is much more challenging to keep private ***and*** transparent records and history of interactions in this way.

Reply:
I agree transparency is important. If the whole committee is supposed to be involved right from the start I think a separate e-mail address like `PySal-conduct@gmail.com` linked to all committee members in addition to a public Gitter channel is a good idea. 

I still think a private Gitter message would be a somewhat `easy` solution to give options for first quick personal contacts. This could even be beneficial for minorities who want to talk to a peer first, e.g. women feeling more comfortable sharing concerns with other women. We could definitely add a short paragraph that states how concerns addressed in private may be shared with the whole committee or community in order to resolve conflicts.

Therefore I'd suggest to:

1. Have an e-mail address connected to all committee members as the recommended way of contact
2. Having a public CoC Gitter channel for further discussion of concerns and including discussions in the 'public'
3. Give the option to talk to specific peers one on one if concerns arise, being aware fo the fact that to resolve conflicts information will be shared with all committee members. 

I think that would provide a platform for a diverse crowd to raise concerns without making processes intransparent. I think we do have to provide a more "personal" option for contact, if all conflicts would be solvable in public we would not need a committee anyway. And I do think providing a more "personal" option would allow us to also cover minorities who might not be as comfortable to speak to anyone on the committee in a first instance. Let me know what you think?

Reply:
@slumnitz Agreed, and good points.

So now we need to set up:
  1. the contact email; and
  2. the gitter room

Reply:
We could create a `conduct@pysal.org` e-mail address via a private alias in the pysal google groups? Someone with admin rights over the pysal.org domain would need to do that.

Reply:
agreed. And I am trying to figure out if we can create a gitter room associated with pysal without have an associated repo.